### PR TITLE
Update get project endpoint

### PR DESCRIPTION
**Pull request by KostinP**
Update endpoint get project via new technical description
http://outsourcing.nat.tepkom.ru:20080/redmine/issues/413

### CODE REVIEW

File: src/ProjectService.Business/Commands/GetProjectByIdCommand.cs [hunk 1]
```diff
@@ -1,32 +1,132 @@
...
+            DepartmentInfo department = null;
+
+            try
+            {
+                var departmentResponse = _departmentRequestClient.GetResponse<IOperationResult<IGetDepartmentResponse>>(
+                    IGetDepartmentRequest.CreateObj(null, departmentId)).Result;
+
+                if (departmentResponse.Message.IsSuccess)
+                {
+                    //TODO Add mapper
```
[the remark this answers is not in the document]
Comment (KostinP) [reply src/ProjectService.Business/Commands/GetProjectByIdCommand.cs:42]: Not edit. Add mapper for map from IGetDepartmentResponse to DepartmentInfo

File: src/ProjectService.Broker/GetProjectInfoConsumer.cs [hunk 2]
```diff
@@ -15,7 +16,8 @@ public class GetProjectInfoConsumer : IConsumer<IGetProjectRequest>
 
         private object GetProjectInfo(Guid projectId)
         {
-            var dbProject = _repository.GetProject(projectId);
+            var filter = new GetProjectFilter { ProjectId = projectId };
```
[the remark this answers is not in the document]
Comment (KostinP) [reply src/ProjectService.Broker/GetProjectInfoConsumer.cs:19]: Repository was changed

File: src/ProjectService.Mappers/ResponsesMappers/ProjectExpandedResponseMapper.cs [hunk 3]
```diff
@@ -1,65 +1,44 @@
-﻿using LT.DigitalOffice.Broker.Requests;
-using LT.DigitalOffice.Broker.Responses;
-using LT.DigitalOffice.Kernel.Broker;
-using LT.DigitalOffice.ProjectService.Mappers.ModelsMappers.Interfaces;
+﻿using LT.DigitalOffice.ProjectService.Mappers.ModelsMappers.Interfaces;
```
Comment (djino97) [inline src/ProjectService.Mappers/ResponsesMappers/ProjectExpandedResponseMapper.cs:1]: need rename mapper by new scheme

File: src/ProjectService.Models.Dto/Requests/Filters/GetProjectFilter.cs [hunk 4]
```diff
@@ -0,0 +1,20 @@
+﻿using Microsoft.AspNetCore.Mvc;
+using System;
+
+namespace LT.DigitalOffice.ProjectService.Models.Dto.Requests.Filters
+{
+    public class GetProjectFilter
+    {
+        [FromQuery(Name = "projectId")]
```
Comment (djino97) [inline src/ProjectService.Models.Dto/Requests/Filters/GetProjectFilter.cs:8]: Fix format query

File: src/ProjectService.Business/Commands/GetProjectByIdCommand.cs [hunk 5]
```diff
@@ -1,32 +1,132 @@
...
+using MassTransit;
+using Microsoft.Extensions.Logging;
 using System;
-using System.Threading.Tasks;
+using System.Collections.Generic;
+using System.Linq;
 
 namespace LT.DigitalOffice.ProjectService.Business.Commands
 {
     public class GetProjectByIdCommand : IGetProjectByIdCommand
```
Comment (BelehovEgor) [inline src/ProjectService.Business/Commands/GetProjectByIdCommand.cs:21]: Can you name it GetProjectCommand?

File: src/ProjectService.Mappers/ResponsesMappers/ProjectExpandedResponseMapper.cs (hunk 3 above)
Comment (djino97) [inline src/ProjectService.Mappers/ResponsesMappers/ProjectExpandedResponseMapper.cs:1]: Simple ProjectResponseMapper

File: src/ProjectService.Business/Commands/GetProjectByIdCommand.cs (hunk 1 above)
[the remark this answers is not in the document]
Comment (BelehovEgor) [reply src/ProjectService.Business/Commands/GetProjectByIdCommand.cs:42]: and why you can't do it?

File: src/ProjectService/Controllers/ProjectController.cs [hunk 6]
```diff
@@ -26,10 +26,9 @@ public class ProjectController : ControllerBase
         [HttpGet("getProjectById")]
```
Comment (djino97) [inline src/ProjectService/Controllers/ProjectController.cs:26]: simple get

File: src/ProjectService/Controllers/ProjectController.cs [hunk 7]
```diff
@@ -26,10 +26,9 @@ public class ProjectController : ControllerBase
...
         public ProjectExpandedResponse GetProjectById(
             [FromServices] IGetProjectByIdCommand command,
```
Comment (djino97) [inline src/ProjectService/Controllers/ProjectController.cs:28]: simple Get

File: src/ProjectService.Models.Dto/Responses/ProjectExpandedResponse.cs [hunk 8]
```diff
@@ -7,7 +7,6 @@ namespace LT.DigitalOffice.ProjectService.Models.Dto.Responses
     public class ProjectExpandedResponse
     {
```
Comment (djino97) [inline src/ProjectService.Models.Dto/Responses/ProjectExpandedResponse.cs:8]: simple ProjectResponse

File: src/ProjectService.Data/UserRepository.cs [hunk 9]
```diff
@@ -17,6 +19,18 @@ public UserRepository(IDataProvider provider)
             _provider = provider;
         }
 
+        public IEnumerable<DbProjectUser> GetProjectUsers(Guid projectId, bool showNotActive)
+        {
+            var predicate = PredicateBuilder.New<DbProjectUser>(u => u.ProjectId == projectId && u.IsActive);
+
+            if (showNotActive)
+            {
+                predicate.Or(u => !u.IsActive);
```
Comment (BelehovEgor) [inline src/ProjectService.Data/UserRepository.cs:28]: Are you sure about this predicate? it looks like u.ProjectId == projectId && u.IsActive || !u.IsActive, so if user is inactive and his id not equal to porjectId we took it too

File: src/ProjectService.Models.Broker/Responses/IGetUsersDataResponse.cs [hunk 10]
```diff
@@ -0,0 +1,18 @@
+﻿using LT.DigitalOffice.Broker.Models;
+using System.Collections.Generic;
+
+namespace LT.DigitalOffice.Broker.Responses
+{
+    public interface IGetUsersDataResponse
+    {
+        List<UserData> UsersData { get; }
```
Comment (BelehovEgor) [inline src/ProjectService.Models.Broker/Responses/IGetUsersDataResponse.cs:8]: maybe better use IList

File: src/ProjectService.Business/Commands/GetProjectByIdCommand.cs [hunk 11]
```diff
@@ -1,32 +1,132 @@
...
+using LT.DigitalOffice.ProjectService.Models.Dto.Models;
+using LT.DigitalOffice.ProjectService.Models.Dto.Models.ProjectUser;
+using LT.DigitalOffice.ProjectService.Models.Dto.Requests.Filters;
 using LT.DigitalOffice.ProjectService.Models.Dto.Responses;
+using MassTransit;
+using Microsoft.Extensions.Logging;
 using System;
-using System.Threading.Tasks;
+using System.Collections.Generic;
+using System.Linq;
```
Comment (djino97) [inline src/ProjectService.Business/Commands/GetProjectByIdCommand.cs:17]: simple GetProjectCommand

File: src/ProjectService.Models.Dto/Models/ProjectUser/ProjectUserInfo.cs [hunk 12]
```diff
@@ -1,13 +1,17 @@
 using LT.DigitalOffice.ProjectService.Models.Dto.Enums;
-using LT.DigitalOffice.ProjectService.Models.Dto.Models.User;
 using System;
 
 namespace LT.DigitalOffice.ProjectService.Models.Dto.Models.ProjectUser
 {
     public class ProjectUserInfo
```
Comment (BelehovEgor) [inline src/ProjectService.Models.Dto/Models/ProjectUser/ProjectUserInfo.cs:6]: why don't we need ProectId here?

File: test/ProjectService.Broker.UnitTests/GetProjectInfoConsumerTests.cs [hunk 13]
```diff
@@ -116,7 +117,7 @@ public async Task ShouldThrowExceptionWhenProjectIdWasNotFound()
                 SerializerAssert.AreEqual(expectedResult, response.Message);
                 Assert.True(_consumerTestHarness.Consumed.Select<IGetProjectRequest>().Any());
                 Assert.True(_harness.Sent.Select<IOperationResult<IProjectResponse>>().Any());
-                _repository.Verify(x => x.GetProject(_projectId), Times.Once);
+                _repository.Verify(x => x.GetProject(It.IsAny<GetProjectFilter>()), Times.Once);
```
Comment (BelehovEgor) [inline test/ProjectService.Broker.UnitTests/GetProjectInfoConsumerTests.cs:120]: I don't think that using It.IsAny in verify  is good

File: test/ProjectService.Business.UnitTests/Commands/EditProjectByIdCommandTests.cs [hunk 14]
```diff
@@ -16,13 +17,15 @@ namespace LT.DigitalOffice.ProjectServiceUnitTests.Commands
 {
     internal class EditProjectByIdCommandTests
     {
-        private DbProject dbProject;
-        private EditProjectRequest editRequest;
         private IEditProjectByIdCommand command;
```
Comment (BelehovEgor) [inline test/ProjectService.Business.UnitTests/Commands/EditProjectByIdCommandTests.cs:20]: use _

**Review by BelehovEgor — CHANGES_REQUESTED** [review]

File: src/ProjectService.Business/Commands/GetProjectByIdCommand.cs [hunk 15]
```diff
@@ -1,32 +1,132 @@
...
+            {
+                List<Guid> userIds;
+
+                if (showNotActiveUsers == true)
+                {
+                    userIds = projectUsers.Select(x => x.Id).Distinct().ToList();
+                }
+                else
+                {
+                    userIds = projectUsers.Where(x => x.IsActive == true).Select(x => x.Id).Distinct().ToList();
```
[the remark this answers is not in the document]
Comment (KostinP) [reply src/ProjectService.Business/Commands/GetProjectByIdCommand.cs:74]: Bug, not id, UserId

File: src/ProjectService.Mappers/ResponsesMappers/ProjectExpandedResponseMapper.cs (hunk 3 above)
Comment (KostinP) [reply src/ProjectService.Mappers/ResponsesMappers/ProjectExpandedResponseMapper.cs:1]: Why?

File: src/ProjectService.Models.Dto/Responses/ProjectExpandedResponse.cs (hunk 8 above)
Comment (KostinP) [reply src/ProjectService.Models.Dto/Responses/ProjectExpandedResponse.cs:8]: Why? ProjectResponse already exist

Comment (djino97) [reply src/ProjectService.Models.Dto/Responses/ProjectExpandedResponse.cs:8]: see this as look like in UserService

File: src/ProjectService.Mappers/ResponsesMappers/ProjectExpandedResponseMapper.cs (hunk 3 above)
Comment (djino97) [reply src/ProjectService.Mappers/ResponsesMappers/ProjectExpandedResponseMapper.cs:1]: see this as look like in UserService

Comment (KostinP) [reply src/ProjectService.Mappers/ResponsesMappers/ProjectExpandedResponseMapper.cs]: Renamed

**Review by djino97 — APPROVED** [review]

**Review by BelehovEgor — APPROVED** [review]

**Review by TIove — APPROVED** [review]

**Review by MariyaSin — APPROVED** [review]

**Pull request merged**